innovation.game.php

```diff
@@ -7979,6 +7979,36 @@ function stPlayerInvolvedTurn() {
                     }
                 }
                 break;
+
+            // id 155, Artifacts age 5: Boerhavve Silver Microscope
+            case "155N1":
+                $step_max = 2;
+                break;
+
+            // id 156, Artifacts age 5: Principia
+            case "156N1":
+                $step_max = 1;
+                break;
+            
+            // id 159, Artifacts age 5: Barque-Longue La Belle
+            case "159N1":
+                do {
+                    $card = self::executeDraw($player_id, 5, 'board'); // "Draw and meld a 5"
+                    // If the drawn card is not green, repeat this effect.
+                } while($card['color'] == 0 || $card['color'] == 1 || $card['color'] == 3 || $card['color'] == 4);
```

Comment:
How about just checking `$card['color'] != 2`?

Reply:
I did that originally, but it it was never true.  Actually, I did !==.

Reply:
!== is a bit different than != since !== also requires the data to be the same type. It's possible that the type of data being pulled from the database is in a slightly format than a number like 2.

I'd recommend trying != and seeing if that works here!

Reply:
Yup, that works!  Thanks.